### PR TITLE
[ready-for-core-team-review]CRM-16189, added hook to alter deferred revenue items

### PR DESCRIPTION
**Pull request by pradpnayak**
- [CRM-16189: Improve support for Accrual Method bookkeeping](https://issues.civicrm.org/jira/browse/CRM-16189)

### CODE REVIEW

**Comment by pradpnayak** [comment]
Is test failure related to PR? https://test.civicrm.org/job/CiviCRM-Core-PR/10768/

**Comment by pradpnayak** [comment]
jenkins, retest this please

File: CRM/Core/BAO/FinancialTrxn.php [hunk 1]
```diff
@@ -669,7 +669,9 @@ public static function createDeferredTrxn($lineItems, $contributionDetails, $upd
         }
       }
       $accountRel = key(CRM_Core_PseudoConstant::accountOptionValues('account_relationship', NULL, " AND v.name LIKE 'Income Account is' "));
-      // TODO: Call hook to alter $deferredRevenues
+
+      CRM_Utils_Hook::alterDeferredRevenueItems($deferredRevenues);
```
Comment (JoeMurray) [inline CRM/Core/BAO/FinancialTrxn.php]: let's add more args here - $contributionDetails, $update, $context - in order to help anyone implementing hook so they know what this function does

File: CRM/Utils/Hook.php [hunk 2]
```diff
@@ -2095,6 +2095,21 @@ public static function batchQuery(&$query) {
...
 
   /**
+   * This hook is called to alter Deferred revenue item values just before they are
+   * inserted in civicrm_financial_trxn table
+   *
+   * @param array $deferredRevenues
+   *
+   * @return mixed
+   */
+  public static function alterDeferredRevenueItems(&$deferredRevenues) {
```
Comment (JoeMurray) [inline CRM/Utils/Hook.php]: change to handle new extra args

**Comment by pradpnayak** [comment]
Added the changes

**Pull request merged**